lib/vizkit/vizkit_items.rb

```diff
@@ -803,6 +803,27 @@ def write(&block)
         end
     end
 
+    class TaskModelItem < VizkitItem
+        attr_reader :model
+
+        def initialize(task, options = Hash.new)
```

Comment:
> Similar for caching options and raising exception -- just maintaining previous behaviour.

Since nobody has the time, effort and/or will to do the necessary work to remove all of them, upgrading bit-by-bit is basically the only thing that would ensure that we may one day have removed the old structure.

Yes, it's inconsistent. But the alternative is to keep on with the old form until the end of Rock.

As for @options, in this particular class @options is not needed. The other items that do cache @options use it, but you don't.